app_feup/lib/model/entities/lecture.dart

```diff
@@ -21,28 +21,70 @@ class Lecture {
   int day;
   int blocks;
   int startTimeSeconds;
+  int occurrId;
 
-  Lecture(String subject, String typeClass, int day, int startTimeSeconds,
-      int blocks, String room, String teacher, String classNumber) {
+  Lecture(
+      String subject,
+      String typeClass,
+      int day,
+      int blocks,
+      String room,
+      String teacher,
+      String classNumber,
+      int startTimeHours,
+      int startTimeMinutes,
+      int endTimeHours,
+      int endTimeMinutes,
+      [int occurrId]) {
     this.subject = subject;
     this.typeClass = typeClass;
     this.room = room;
     this.teacher = teacher;
     this.day = day;
     this.blocks = blocks;
-    this.startTimeSeconds = startTimeSeconds;
     this.classNumber = classNumber;
+    this.occurrId = occurrId;
 
-    this.startTime = (startTimeSeconds ~/ 3600).toString().padLeft(2, '0') +
-        ':' +
-        ((startTimeSeconds % 3600) ~/ 60).toString().padLeft(2, '0');
-    startTimeSeconds += 60 * 30 * blocks;
-    this.endTime = (startTimeSeconds ~/ 3600).toString().padLeft(2, '0') +
-        ':' +
-        ((startTimeSeconds % 3600) ~/ 60).toString().padLeft(2, '0');
+    this.startTime = startTimeHours.toString().padLeft(2, '0') +
+        'h' +
+        startTimeMinutes.toString().padLeft(2, '0');
+    this.endTime = endTimeHours.toString().padLeft(2, '0') +
+        'h' +
+        endTimeMinutes.toString().padLeft(2, '0');
+  }
```

Comment:
Introduced old changes from on the hours are displayed, make sure to switch them back to `:` instead of `h` as the hour separator